pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,426 @@
+#
+#
+"""
+A DMTF CIM schema is the collection of CIM qualifier declarations and CIM
+classes managed by the DMTF and released as a single tested, coherent package
+with a defined major, minor, update version number.  The MOF representation of
+a DMTF schema release is packaged as a single zip file and is available on the
+DMTF web site.
+
+Because the DMTF MOF schemas are the source of most of the classes that a WBEM
+server would implement we felt it was important to create a simple mechanism
+to include the DMTF CIM classes and qualifier declarations in the
+:class:`~pywbem.FakedWBEMConnection` mock repository.
+
+The :class:`~pywbem_mock.DMTFCIMSchema` class downloads the DMTF CIM schema MOF
+(a zip file) defined by version from the DMTF web site into a defined directory,
+unzips the MOF files into a subdirectory and makes the this subdirectory
+available as a property of the :class:`~pywbem_mock.DMTFCIMSchema` object.
+
+Multiple DMTF CIM schemas may be maintained in the same `schema_root_dir`
+simultaneously because each schema is expanded into a subdirectory identified
+by the schema version information.
+
+NOTE: This class works only for the DMTF repository because it gets the
+original zip file from the DMTF repository with the particular url and zip
+file organization of the DMTF CIM Schema release.
+
+The implementation of this class also depends on all of the zip files in
+the CIM schema having unique class names, which is always true for DMTF
+CIM schema releases to assure that they can be installed in the same
+namespace in a WBEM server repository.
+
+Once a DMTF CIM schema is expanded into the individual schema files it consumes
+considerable space since it expands to almost 3000 files.  Therefore it is
+logical to remove all of the individual MOF files when not being used and also
+if the schema information is to be committed to be used in testing in an
+environment like github. Therefore, a method
+:meth:`~pywbem_mock.DMTFCIMSchema.clean` removes all of the MOF files from
+local storage.  They are restored the next time the
+:class:`~pywbem_mock.DMTFCIMSchema` object for the same version is constructed.
+
+The DMTF maintains two versions of each released schema:
+
+* Final - This schema does not contain any of the schema classes that are
+  marked experimantel.  It is considered the stable release.
+
+* Experimental - Contains the classes in the `Final` schema file plus other
+  classes that are considered experimental. Typically it is considered less
+  stable.
+
+:class:`~pywbem_mock.DMTFCIMSchema` can install either the `Final` or
+'experimental' schema depending on the value of the `user-experimental`
+parameter.
+
+The :class:`~pywbem_mock.DMTFCIMSchema` class also provides a
+:meth:`~pywbem_mock.build_schema_mof` to create a MOF include file from the
+DMTF Schema downloaded that includes only a subset of the classes actually in
+the repository for compile into a MOF repository.
```

Comment:
That sentence can be improved. Change its second part to: "... to create a MOF include file that includes only a subset of the classes in the downloaded DMTF CIM schema. This is useful if only a subset of the classes is supposed to be compiled into the mock repository."

Reply:
Did it slightly different by putting the why part at the head of the sentence 

DONE

Because it is usually necessary to compile only a subset of the DMTF CIM
classes into the mock repository for any test, the
:class:`~pywbem_mock.DMTFCIMSchema` class provides a
:meth:`~pywbem_mock.build_schema_mof` method to create a MOF include file that
includes only a subset of the classes in the downloaded DMTF CIM Schema.